src/com/facebook/buck/json/BuildFilePythonResultDeserializer.java

```diff
@@ -110,7 +110,21 @@ public BuildFilePythonResult deserialize(JsonParser jp, DeserializationContext c
     ImmutableList.Builder<Object> builder = ImmutableList.builder();
     JsonToken token;
     while ((token = jp.nextToken()) != JsonToken.END_ARRAY) {
-      builder.add(deserializeRecursive(jp, token));
+      Object obj = deserializeRecursive(jp, token);
+      if (obj != null) {
+        builder.add(obj);
+      }
+      else {
+        // null elements can't be added to ImmutableList, an NPE will be thrown.
+        // Throw a meaningful exception here instead.
+        StringBuilder message = new StringBuilder();
+        message.append("null value can't be added to [");
+        for (Object element: builder.build()) {
+          message.append("'").append(element).append("'").append(", ");
+        }
+        message.append("]");
+        throw new JsonParseException(null, message.toString());
```

Comment:
This isn't really a JSON parse exception, is it? It sounds like the JSON is valid, just that we can't actually do anything with null.

Reply:
Changed to IllegalArgumentException.